### DISCUSSION TITLE
Naming conventions - What are allowed names for namespace and folders in Joomla?

### DISCUSSION THREAD

**Discussion by astridx** [joomla/joomla-cms#36679] [Q&A]
I have a problem with a view name. I like to use `FOOPlaces`. 
The view is not found using this name.

After I renamed `view`-`folder` and `namespace` to `Fooplaces` everything works fine.

I found this site. 
https://docs.joomla.org/J4.x:File_Structure_and_Naming_Conventions

According to this site the foldername for the template should use lower case. 

**What are allowed names for namespace and folders in Joomla?**

**Comment by Fedik** [comment]
If you have a view `FOOPlaces`, then:
File to the view class: `patch_to_component/View/FOOPlaces/HtmlView.php`
Full class name with namsepace (for `Site`): `Astridx\Component\MyCoolComponent\Site\View\FOOPlaces\HtmlView`
And `tmpl` file:  `patch_to_component/tmpl/fooplaces/default.php`

Can use Content Article view as example [components/com_content/src/View/Article/HtmlView.php](https://github.com/joomla/joomla-cms/blob/4.0-dev/components/com_content/src/View/Article/HtmlView.php)

UPD:

> After I renamed view-folder and namespace to Fooplaces everything works fine.

Ah, maybe that because in URL it willl be `view=fooplaces`, and so you can use only `Fooplaces`.
Yeah, looks like a litle limitation.
